indexers/dbms/BinaryPbIndexer/__init__.py

```diff
@@ -0,0 +1,273 @@
+import shutil
+from typing import List, Tuple, Generator, Optional, Iterable
+import numpy as np
+import os
+from jina import Executor, requests, DocumentArray, Document
+from jina.logging import JinaLogger
+
+from jina.executors.indexers.dump import export_dump_streaming
+from jina.helper import call_obj_fn, cached_property, get_readable_size
+from jina.executors.helper import physical_size
+from .binarypb import BinaryPbWriterMixin
+
+# from .query import BinaryPbQueryIndexer
+
+HEADER_NONE_ENTRY = (-1, -1, -1)
+
+
+class BinaryPbDBMSIndexer(Executor, BinaryPbWriterMixin):
+    """
+    A DBMS Indexer (no query method)
+
+    :param index_filename: the name of the file for storing the index, when not given metas.name is used.
+    :param key_length: the default minimum length of the key, will be expanded one time on the first batch
+    :param args:  Additional positional arguments which are just used for the parent initialization
+    :param kwargs: Additional keyword arguments which are just used for the parent initialization
+    """
+
+    def __init__(
+        self,
+        index_filename: Optional[str] = None,
+        key_length: int = 36,
+        dump_on_exit: str = True,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.index_filename = index_filename or self.metas.name
+
+        self.key_length = key_length
+
+        self.handler_mutex = True
+        self.is_handler_loaded = False
+
+        self._page_size = mmap.ALLOCATIONGRANULARITY
+        self.logger = JinaLogger(self.__class__.__name__)
+        self._dump_on_exit = dump_on_exit
+
+        if self.is_exist:
+            self._size = self.query_handler.size()
+            self._start = self.query_handler.total_bytes
+        else:
+            self._size = 0
+            self._start = 0
+
+    @cached_property
+    def query_handler(self):
+        """A readable and indexable object, could be dict, map, list, numpy array etc.
+
+        :return: read handler
+
+        .. note::
+            :attr:`query_handler` and :attr:`write_handler` are by default mutex
+        """
+        r = None
+        if not self.handler_mutex or not self.is_handler_loaded:
+            r = self.get_query_handler()
+            if r is None:
+                self.logger.warning(
+                    f'you can not query from {self} as its "query_handler" is not set. '
+                    'If you are indexing data from scratch then it is fine. '
+                    'If you are querying data then the index file must be empty or broken.'
+                )
+            else:
+                self.logger.info(f'indexer size: {r.size()}')
+                self.is_handler_loaded = True
+        return r
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    @property
+    def is_exist(self) -> bool:
+        """
+        Check if the database is exist or not
```

Comment:
phrasing is off. I know it was there before, but now it's the time to change it :)

```suggestion
        Check if the database exists or not
```